pkg/api/customization/cluster/validator.go

```diff
@@ -48,17 +53,29 @@ func (v *Validator) Validator(request *types.APIContext, schema *types.Schema, d
 		return err
 	}
 
-	if err := validateScheduledClusterScan(&spec); err != nil {
+	if err := v.validateScheduledClusterScan(&spec); err != nil {
 		return err
 	}
 	return nil
 }
 
-func validateScheduledClusterScan(spec *v3.ClusterSpec) error {
+func (v *Validator) validateScheduledClusterScan(spec *v3.ClusterSpec) error {
 	if spec.ScheduledClusterScan == nil ||
 		(spec.ScheduledClusterScan != nil && !spec.ScheduledClusterScan.Enabled) {
 		return nil
 	}
+	currentK8sVersion := spec.RancherKubernetesEngineConfig.Version
+	overrideBenchmarkVersion := ""
+	if spec.ScheduledClusterScan.ScanConfig.CisScanConfig != nil {
+		overrideBenchmarkVersion = spec.ScheduledClusterScan.ScanConfig.CisScanConfig.OverrideBenchmarkVersion
+	}
+	_, _, err := cis.GetBenchmarkVersionToUse(overrideBenchmarkVersion, currentK8sVersion,
+		v.CisConfigLister, v.CisConfigClient,
+		v.CisBenchmarkVersionLister, v.CisBenchmarkVersionClient,
+	)
+	if err != nil {
+		return err
```

Comment:
We should return a HTTP error in here

Reply:
I will return a http error, the message is already coming from the function being called.